solver/solver.go

```diff
@@ -125,6 +125,13 @@ func (s *Solver) SolveRequest(ctx context.Context, req bkgw.SolveRequest) (*bkgw
 	// makes Solve() to block until LLB graph is solved. otherwise it will
 	// return result (that you can for example use for next build) that
 	// will be evaluated on export or if you access files on it.
+	//
+	// NOTE: if a future change modifies Evaluate to not always be true anymore, we
+	// will need to ensure that Solver.Export no longer sets "ignoreLogs" to true
+	// when forwarding progress events. This is because those logs will no longer
+	// necessarily be duped from the main channel published to by Solves called here.
+	// Sample code to properly handle that can be found here:
+	// https://github.com/sipsma/dagger/commit/104e0d0393b5f707ea40448736f2e0e87fb1e4ed
 	req.Evaluate = true
```

Comment:
I think for now this is the right approach because it keeps things simpler for the reasons you said. The only downside I can think of is that it's possible Buildkit's solver won't be able to parallelize as effectively when vertex DAGs are sent one by one, with each solve blocking due to `Evaluate` being true.

However, we have the `cue/flow` solver layered on top of the Buildkit solver and it knows how to schedule vertices on the "task level" efficiently. As long as we have a 1-to-1 correspondence between LLB vertices and cue/flow tasks this should be good enough I think. If we ever deviate from that model significantly (i.e. we end up with cue tasks that create a bunch of LLB vertices that don't correspond to other cue tasks) then this could be worth revisiting.

This is a subtle issue though, these are just my immediate thoughts, worth considering more.